src/signals/incident-management/containers/IncidentSplitContainer/IncidentSplitContainer.js

```diff
@@ -26,32 +27,78 @@ const getParentIncident = incident => ({
 });
 
 const IncidentSplitContainer = ({ FormComponent }) => {
-  const { data, error, get, isLoading, isSuccess, post } = useFetch();
+  const { error: errorSplit, isSuccess: isSuccessSplit, post } = useFetch();
+  const {
+    data: dataParent,
+    error: errorParent,
+    get: getParent,
+    isLoading: isLoadingParent,
+    isSuccess: isSuccessParent,
+  } = useFetch();
+  const { error: errorUpdate, patch, isSuccess: isSuccessUpdate } = useFetch();
   const { id } = useParams();
   const history = useHistory();
   const dispatch = useDispatch();
-
+  const [parentIncident, setParentIncident] = useState();
+  const [directingDepartment, setDirectingDepartment] = useState([]);
+  const departments = useSelector(makeSelectDepartments);
   const subcategories = useSelector(makeSelectSubCategories);
 
   const subcategoryOptions = useMemo(
     () => subcategories?.map(category => ({ ...category, value: category.extendedName })),
     [subcategories]
   );
 
+  const updateDepartment = useCallback(
+    name => {
+      if (!name || !departments) return;
+
+      const department = departments.list.find(d => d.code === name);
+      setDirectingDepartment(department ? [{ id: department.id }] : []);
+    },
+    [departments, setDirectingDepartment]
+  );
+
   useEffect(() => {
-    get(`${configuration.INCIDENT_PRIVATE_ENDPOINT}${id}`);
-  }, [get, id]);
+    getParent(`${configuration.INCIDENT_PRIVATE_ENDPOINT}${id}`);
+  }, [getParent, id]);
 
   useEffect(() => {
-    if (isSuccess === undefined || error === undefined) return;
+    if (isLoadingParent || isSuccessParent || errorParent === undefined) return;
 
-    const notificationProps = isSuccess
+    if (errorParent === false) {
+      setParentIncident(dataParent);
+    }
+  }, [errorParent, isLoadingParent, isSuccessParent, dataParent]);
+
+  useEffect(() => {
+    if (isSuccessSplit === undefined || errorSplit === undefined) return;
+    if (isSuccessSplit) {
+      patch(`${configuration.INCIDENT_PRIVATE_ENDPOINT}${id}`, { directing_department: directingDepartment });
+    } else {
+      dispatch(
+        showGlobalNotification({
+          title: 'De melding kon niet gesplitst worden',
+          variant: VARIANT_ERROR,
+          type: TYPE_LOCAL,
+        })
+      );
+      history.push(`${INCIDENT_URL}/${id}`);
+    }
+
+    // Disabling linter; patch and history dependencies are generating infinite loops
```

Comment:
This shouldn't be the case, certainly not for `patch`, because that is a memoized function in the `useFetch` hook. Which dependency causes the loop?

Reply:
both history and patch

Reply:
Might be necessary to change the `useFetch` hook so that the returns functions are memoized:
```javascript
const post = useMemo(() => modify('POST'), [modify]);
const patch = useMemo(() => modify('PATCH'), [modify]);
const put = useMemo(() => modify('PUT'), [modify]);
```

Reply:
I'll look into that

Reply:
It helps, but I still have a problem with the history. It's generating also the infinite loop. I could memoize that but it doesn't fill right...

Reply:
This might be related: https://github.com/ReactTraining/react-router/issues/7000

Reply:
It looks like, indeed. I'll eliminate the `history` the dependency for now.